File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        minValue = float('inf')  # Start with a very large value
        maxProfit = 0  # To store the maximum profit

        array = []
        minV = 0

        # Find the minimum value and its index
        for i in range(len(prices)):
            if prices[i] < minValue:
                minValue = prices[i]
                minV = i  # Update the index of the minimum value

        # Collect all prices after the minimum value
        for i in range(minV, len(prices)):
            if prices[i] > minValue:
                array.append(prices[i])

        # Calculate the maximum profit
        if array:
            maxValue = max(array)
            maxProfit = maxValue - minValue

        return maxProfit

                


        



        

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) 
    {
        int res = INT_MAX;
        int profit = 0;

        for (int i{}; i < prices.size(); i++)
        {
            //Find min
            res = std::min(res, prices[i]);

            //Calc price
            profit = std::max(profit, prices[i] - res);

        }

        return profit > 0 ? profit : 0;
    }
};